cmake/partition_manager.cmake

```diff
@@ -23,8 +23,16 @@ endmacro()
 
 # Load static configuration if found.
 set(user_def_pm_static ${PM_STATIC_YML_FILE})
-set(nodomain_pm_static ${APPLICATION_CONFIG_DIR}/pm_static.yml)
-set(domain_pm_static ${APPLICATION_CONFIG_DIR}/pm_static_${DOMAIN}.yml)
+if(EXISTS ${APPLICATION_CONFIG_DIR}/pm_static_${CONF_FILE_BUILD_TYPE}.yml)
+  set(nodomain_pm_static ${APPLICATION_CONFIG_DIR}/pm_static_${CONF_FILE_BUILD_TYPE}.yml)
+else()
+  set(nodomain_pm_static ${APPLICATION_CONFIG_DIR}/pm_static.yml)
+endif()
+if(EXISTS ${APPLICATION_CONFIG_DIR}/pm_static_${CONF_FILE_BUILD_TYPE}_${DOMAIN}.yml)
+  set(nodomain_pm_static ${APPLICATION_CONFIG_DIR}/pm_static_${CONF_FILE_BUILD_TYPE}_${DOMAIN}.yml)
+else()
+  set(domain_pm_static ${APPLICATION_CONFIG_DIR}/pm_static_${DOMAIN}.yml)
+endif()
```

Comment:
This should already be handled by this code in `ncs_file()` which is called below:
https://github.com/nrfconnect/sdk-nrf/blob/a96e84fdb45ca1135e23b0d6293e69f2829fffbd/cmake/extensions.cmake#L233-L253

We should investigate why that is not working as expected.
Do we have an error or is there a use case we didn't consider in first version of code.

Reply:
Will investigate. This seems to be the only problem - everything else worked like a charm.

Reply:
I think the problem is happening because `ncs_file` is called only on `board_pm_static` and `board_def_pm_static` and these are unset. The variable that is set is nodomain but it does not depend on buildtype.

Reply:
`ncs_file` looks for the following files:
`nrf52840dk_nrf52840_b0`
`.../ncs/nrf/samples/bootloader/pm_static_nrf52840dk_nrf52840_b0.yml`

`../ncs/zephyr/boards/arm/nrf52840dk_nrf52840/pm_static_nrf52840dk_nrf52840_b0.yml`

All seem wrong to me. Since we already pointed to the configuration folder, and by default it looks for `pm_static.yml ` it is logical that it should pick up `pm_static_buildtype`.

Reply:
But it seems that the function always adds the `nrf52840dk_nrf52840` somewhere to the file name (I expect it is not aware that is already at board specific configuration directory.
The `pm_static.yml` is considered only because `partition_manager.cmake` explicitly checks it. It may be best to check for build type at the same place.

Reply:
how about following?

```
<pre><b>diff --git a/cmake/extensions.cmake b/cmake/extensions.cmake</b>
<b>index 701ca1fee..fed621d03 100644</b>
<b>--- a/cmake/extensions.cmake</b>
<b>+++ b/cmake/extensions.cmake</b>
<font color="#06989A">@@ -235,9 +235,16 @@</font> Please provide one of following: CONF_FILES&quot;)
                         BUILD ${ZEPHYR_FILE_BUILD}
     )
     list(APPEND filename_list ${filename})
<font color="#4E9A06">+</font>
<font color="#4E9A06">+    zephyr_build_string(filename</font>
<font color="#4E9A06">+                        BUILD ${ZEPHYR_FILE_BUILD}</font>
<font color="#4E9A06">+    )</font>
<font color="#4E9A06">+    list(APPEND filename_list ${filename})</font>
     list(REMOVE_DUPLICATES filename_list)
 
     foreach(filename ${filename_list})
<font color="#4E9A06">+      string(REGEX REPLACE &quot;^_&quot; &quot;&quot; filename &quot;${filename}&quot;)</font>
<font color="#4E9A06">+</font>
       if(DEFINED NCS_FILE_DOMAIN)
         if(EXISTS ${NCS_FILE_CONF_FILES}/pm_static_${filename}_${NCS_FILE_DOMAIN}.yml)
           set(${NCS_FILE_PM} ${NCS_FILE_CONF_FILES}/pm_static_${filename}_${NCS_FILE_DOMAIN}.yml PARENT_SCOPE)</pre>
```

Reply:
This will also traverse files with pure build_type sufix. The problem is that the zephyr build string adds leading underscore in this case.

Reply:
Hi @tejlmand , I have pushed the change proposed above. See if it's acceptable

Reply:
@tejlmand To sum up this part. Do you prefer to use the original idea with `if(EXISTS ${APPLICATION_CONFIG_DIR}/pm_static_${CONF_FILE_BUILD_TYPE}.yml)` instead of the expansion in `ncs_file`?

Reply:
> Do you prefer to use the original idea with if(EXISTS ${APPLICATION_CONFIG_DIR}/pm_static_${CONF_FILE_BUILD_TYPE}.yml) instead of the expansion in ncs_file?

sorry, didn't notice you had actually pushed the change also.

The `ncs_file` approach does look a clean direction but we must make sure it's clear that it behaves differently than `zephyr_file()`.

Also, if we take the alternative approach we should adjust those names:
https://github.com/nrfconnect/sdk-nrf/blob/74adc1c3a2b577d15df33164bf40fe860fba3f2f/cmake/partition_manager.cmake#L30
https://github.com/nrfconnect/sdk-nrf/blob/74adc1c3a2b577d15df33164bf40fe860fba3f2f/cmake/partition_manager.cmake#L36

as they are not exactly board specific anymore.

Will allow myself to consider both approaches today.

Reply:
I will prepare `ncs_file` change then. Will migrate the default fallback there too (pm_static.yml). This way we will have a single source of configuration file name for partition manager.

Will get back in a minute.

Reply:
@tejlmand , can you check now?